sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4461,6 +4461,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val ALLOW_INTERVAL_TYPES_IN_VARIANT = buildConf("spark.sql.variant.allowIntervalTypes")
```

Comment:
I think maybe making this a conf might be dangerous. We should just never create a variant value that contains an interval, since interval is not a valid type in variant right now.

It is possible that a user enables this, writes a variant that contains this version of interval, but later a different version of an interval is updated in the spec. Then, there would be corrupt variant values.

Reply:
I have removed all interval code now.